invokeai/app/api/routers/models.py

```diff
@@ -54,6 +55,90 @@ async def list_models() -> ModelsList:
     models = parse_obj_as(ModelsList, { "models": models_raw })
     return models
 
+# Kent wrote the below code. It is highly suspect, and should be reviewed. Once all issues have been identified and fixed, this comment can be removed.
+# Seriously, check this. I have no idea how to test it.
+
+""" Add Model """
+@models_router.post(
+    "/",
+    operation_id="add_model",
+    responses={201: {"model": Union[CkptModelInfo, DiffusersModelInfo], "new_model_list": ModelsList}},
+)
+async def add_model(
+    model_info: Union[CkptModelInfo, DiffusersModelInfo],
```

Comment:
You need to use a discriminated union to tell which type the user passed:
```py
Annotated[Union[(CkptModelInfo,DiffusersModelInfo)], Field(discriminator="format")]
```
Though see below - I think you need a wrapping object here.